client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,177 @@
 import Button from '@/src/shared/components/Button';
 import Modal from '@/src/shared/components/Modal/Modal';
 import Pagination from '@/src/shared/components/Pagination';
 import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import CloseIcon from '@/src/shared/icons/CloseIcon';
-import { type Learner } from '@/src/shared/utils';
-import React, { useState } from 'react';
-
-interface AddLearnerModalProps {
-  learners: Learner[];
-}
+import { useRouter } from 'next/router';
+import { useDispatch } from 'react-redux';
+import type { AppDispatch } from '@/app/store';
+import { storeEnrollee } from '@/services/posts/addNewEnrollees';
 
-const AddLearnerModal: React.FC<AddLearnerModalProps> = ({ learners }: AddLearnerModalProps): JSX.Element => {
+const AddLearnerModal: React.FC = (): JSX.Element => {
+  const router = useRouter();
+  const params = router.query;
   const [currentPage, setCurrentPage] = useState(1);
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const courseID = params.id;
+  const dispatch = useDispatch<AppDispatch>();
+  const { data, refetch } = useGetEnrolleesQuery({
+    courseID,
+    search,
+    page,
+  });
 
-  const handleAddLearners = (): void => {
-    alert('Add Learners');
-  };
+  const enrollees = data?.enrollees;
 
   const handleModalClose = (): void => {
+    setSelectedIds([]);
     setIsModalOpen(false);
+    setSearch('');
+    setPage(1);
+    void refetch();
   };
 
-  const handleSearch = (search: string): void => {
-    alert(`You searched ${search}`);
+  const handleAddMaterialModal = (): void => {
+    setSelectedIds([]);
+    setSearch('');
+    setPage(1);
+    void refetch();
+    setCurrentPage(currentPage);
+    setIsModalOpen(!isModalOpen);
   };
 
-  const learnersPerPage = 8;
-  const totalPages = Math.ceil(learners.length / learnersPerPage);
-
-  const lastIndex = currentPage * learnersPerPage;
-  const firstIndex = lastIndex - learnersPerPage;
-  const currentLearners = learners.slice(firstIndex, lastIndex);
+  const handleSearch = (search: string): void => {
+    setSearch(search);
+    setPage(1);
+    void refetch();
+  };
 
   const handleChangePageEvent = (page: number): void => {
     setCurrentPage(page);
+    setPage(page);
+    void refetch();
+  };
+
+  const handleCheckboxChange = (id: number): void => {
+    if (selectedIds.includes(id)) {
+      setSelectedIds(selectedIds.filter((selectedId) => selectedId !== id));
+    } else {
+      setSelectedIds([...selectedIds, id]);
+    }
+  };
+
+  const handleAddLearners = (): void => {
+    const enrollees = selectedIds.join(',');
+    const postData = {
+      trainee: enrollees,
+    };
+    setSelectedIds([]);
+    void dispatch(storeEnrollee({ courseID: params.id, postData }));
+    setSelectedIds([]);
+    handleModalClose();
```

Comment:
```suggestion
    void dispatch(storeEnrollee({ courseID: params.id, postData }));
    handleModalClose();
```
Please remove these 2 `setSelectedIds([]);` duplicates , I think you are already setting `selectedIds` to empty array in `handleModalClose`